lib/wallet/walletdb.js

```diff
@@ -503,6 +503,93 @@ class WalletDB extends EventEmitter {
     }
   }
 
+  /**
+   * Deep Clean:
+   * Keep all keys, account data, wallet maps (name and path).
+   * Dump all TX history and balance state.
+   * A rescan will be required but is not initiated automatically.
+   * @returns {Promise}
+   */
+
+  async deepClean() {
+    const unlock1 = await this.txLock.lock();
+    const unlock2 = await this.writeLock.lock();
+    const unlock3 = await this.readLock.lock();
+    try {
+      return await this._deepClean();
+    } finally {
+      unlock3();
+      unlock2();
+      unlock1();
+    }
+  }
+
+  /**
+   * Deep Clean (without locks):
+   * Keep all keys, account data, wallet maps (name and path).
+   * Dump all TX history and balance state.
+   * A rescan will be required but is not initiated automatically.
+   * @returns {Promise}
+   */
+
+  async _deepClean() {
+    this.logger.warning('Initiating Deep Clean...');
+
+    const b = this.db.batch();
+    const removeRange = (opt) => {
+      return this.db.iterator(opt).each(key => b.del(key));
+    };
+
+    this.logger.warning('Clearing block map, tx map and outpoint map...');
+    // b[height] -> block->wid map
+    await removeRange({
+      gte: layout.b.min(),
+      lte: layout.b.max()
+    });
+
+    // o[hash][index] -> outpoint->wid map
+    await removeRange({
+      gte: layout.o.min(),
+      lte: layout.o.max()
+    });
+
+    // T[hash] -> tx->wid map
+    await removeRange({
+      gte: layout.T.min(),
+      lte: layout.T.max()
+    });
+
+    const wids = await this.db.keys({
+      gte: layout.W.min(),
+      lte: layout.W.max(),
+      parse: key => layout.W.decode(key)[0]
+    });
+
+    for (const wid of wids) {
+      const wallet = await this.get(wid);
+      this.logger.warning(
+        'Clearing all tx history for wallet: %s (%d)',
+        wallet.id, wid
+      );
+
+      // remove all txdb data *except* blinds ('v')
+      const key = 'v'.charCodeAt();
+      const prefix = layout.t.encode(wid);
+      await removeRange({
+        gte: Buffer.concat([prefix, Buffer.alloc(1)]),
```

Comment:
Is this supposed to be a `1` or a `0`?

Reply:
It "should" be a single `0x00` byte buffer. This code will remove everything in the txdb _up to_ the blinds. The next snippet removes everything in the txdb _after_ the blinds:

Removed: `0x00 -> v` and `v -> 0xff`